File: backend/src/main/kotlin/mate4ever/ttip/model/Pet.kt
package mate4ever.ttip.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDate
import javax.validation.constraints.NotEmpty
@Document("pet")
open class Pet(
    @NotEmpty(message = "El nombre es obligatorio")
    var name: String,
    @NotEmpty(message = "La imagen es obligatoria")
    var image: String,
    var birth: LocalDate?,
    @NotEmpty(message = "Debes elegir el tipo de animal que estas publicando")
    var type: String,
    var breed: String?,
    @NotEmpty(message = "Debes elegir en que estado se encuentra el animal")
    var state: String,
    @NotEmpty(message = "El tutor es obligatorio")
//    @ManyToOne(fetch = FetchType.LAZY)
//    @JoinColumn(name = "pets")
    var user: String,
    var vaccine: Boolean,
    var castrated: Boolean,
    var medicalHistory: String?,
    var description: String?,
    var coordinates: Map<String, Double>?
) {
    @Id
    var id: String? = null
}

File: frontend/model/Pet.js
export default class Pet{
    id;
    name;
    image;
    age;
    date;
    type;
    breed;
    state;
    tutor;
    description;

    constructor(pet) {
            this.id = pet.id,
            this.name = pet.name,
            this.image = pet.image,
            this.age = pet.age,
            this.date = pet.date,
            this.type = pet.type,
            this.breed = pet.breed,
            this.state = pet.state,
            this.tutor = pet.user,
            this.description = pet.description;
    }

    isThisState(stateToAsk) {return stateToAsk === this.state}
    isLost() {return this.state === 'Perdido'}
}